src/handlers/records-read.ts

```diff
@@ -65,16 +65,20 @@ export class RecordsReadHandler implements MethodHandler {
 
     const matchedMessage = existingMessages[0];
 
+    // if the matched message is a RecordsDelete, we mark the record as not-found and return both the RecordsDelete the initial write
+    // NOTE: there is an opportunity in the future here to restrict this behavior further:
+    //       ie. perform similar authorization checks as when records exists before returning data about deleted record
```

Comment:
```suggestion
    // if the matched message is a RecordsDelete, we mark the record as not-found and return both the RecordsDelete and the initial RecordsWrite
    // NOTE: there is an opportunity in the future here to restrict this behavior further:
    //       ie. perform similar authorization checks as when records exists before returning data about deleted record
```

Great call out!
Nit: Can we include an issue and TODO? This is definitely one i'd like to see prioritized at some point.

Reply:
Good idea, created issue and added a TODO referencing it.